src/txn/txn_log.c

```diff
@@ -39,14 +39,17 @@ __txn_op_log_row_key_check(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt)
         page = cbt->ref->page;
         WT_ASSERT(session, cbt->slot < page->entries);
         rip = &page->pg_row[cbt->slot];
-        WT_ASSERT(session, __wt_row_leaf_key(session, page, rip, &key, false) == 0);
+        WT_ASSERT_ALWAYS(session, __wt_row_leaf_key(session, page, rip, &key, false) == 0,
+          "Failed to set a buffer for row-store leaf page key");
```

Comment:
"Failed to instantiate a row-store key, cannot proceed with cursor key verification."